src/Observables.jl

```diff
@@ -326,13 +282,6 @@ obsid(observable::AbstractObservable) = obsid(observe(observable))
 listeners(observable::Observable) = observable.listeners
 listeners(observable::AbstractObservable) = listeners(observe(observable))
 
-struct OnUpdate{F, Args} <: Function
-    f::F
-    args::Args
-end
-
-(ou::OnUpdate)(_) = ou.f(map(to_value, ou.args)...)
-
```

Comment:
So, this is a difficult design decision. Specialization adds to latency but improves inference, which might make subsequent code compile faster/smaller/better (e.g., more invalidation-resistant). When I was working on this I decided that the `on` functionality was not going to be inferrable but tried to make `map` inferrable when used appropriately. We can change that, of course.

While it's a gamble, I'm increasingly optimistic that Julia 1.9 might cache native code as well as the type-inferred code. So anything that can be cached (i.e., uses only predictable types) may not cost you latency that much longer.

Reply:
> I'm increasingly optimistic that Julia 1.9 might cache native code 

Oh wow ... That'd change everything I suppose :D